agentendpoint/config_task.go

```diff
@@ -32,9 +33,15 @@ const (
 	postCheckDesiredStateStepIndex = 3
 )
 
+var newResource = func(r *agentendpointpb.ApplyConfigTask_Config_Resource) resourceIface {
```

Comment:
This is an overall comment for the file and not specific to this line.
The word resource appears in multiple contexts and reading for the first time, it is not immediately obvious which resource are we talking about without reading much further. Consider renaming the types so that it is easy to quickly decipher which flavor of resource we are talking about.